noir-projects/aztec-nr/aztec/src/messages/message_delivery.nr

```diff
@@ -0,0 +1,75 @@
+/// Specifies the configuration parameters for message delivery. There are two fundamental aspects to consider:
+///
+/// +----------------------------------------------------------------------------------------------------------+
+/// | 1. Delivery Mechanism                                                                                    |
+/// | - Messages can be delivered either on-chain or out-of-band                                               |
+/// | - On-chain delivery uses the Aztec protocol's private log stream, submitted to L1 blobs and consuming DA |
+/// | - Out-of-band delivery is implemented by the application (e.g. storing ciphertexts in cloud storage)     |
+/// | - Out-of-band delivery cannot have any cryptographic constraints since messages are never stored on-chain|
+/// +----------------------------------------------------------------------------------------------------------+
+///
+/// For on-chain delivery, we must also consider:
+///
+/// +----------------------------------------------------------------------------------------------------------+
+/// | 2. Message Encryption and Tagging                                                                        |
+/// | - Messages can use either constrained or unconstrained encryption                                        |
+/// | - Constrained encryption guarantees the ciphertext is formed correctly but costs more in constraints,    |
+/// |   which results in slower proving times                                                                  |
+/// | - Unconstrained encryption trusts the sender but is cheaper constraint-wise and hence faster to prove    |
+/// | - Tagging is an indexing mechanism that helps recipients locate their messages                           |
+/// | - If tagging is not performed correctly by the sender, the recipient will not be able to find the message|
+/// +----------------------------------------------------------------------------------------------------------+
+///
+/// For off-chain delivery, constrained encryption is not relevant since it doesn't provide any additional guarantees
+/// over unconstrained encryption and is slower to prove (requiring more constraints).
+///
+/// There are three available delivery modes described below.
+// TODO(#16771): This is now used by notes as well. Move it.
```

Comment:
Doesn't this PR address this issue?

Reply:
Forgot to remove it. Thanks for noticing it

Reply:
Will remove in that other PR of mine as I'm AFK now and don't want to then wait 3 hours for it to merge.